tests/_internal/config/test_config.py

```diff
@@ -51,6 +53,20 @@ def test_keeps_old_option_data(self):
 
         self.assertEqual(config.options["bool"].help_text, "A simple bool")
 
+    def test_does_not_explode_on_cumulative_override_keys(self):
+        for key in CUMULATIVE_OVERRIDE_KEYS_IN_SOURCE:
+            with self.subTest("Ignores {key} cumulative override"):
+                source_a = SourceFactory()
+                source_b = SourceFactory(overrides={key: "boom?"})
+                Config.from_sources([source_a, source_b])
+
+    def test_does_not_explode_on_bootloader_override_keys(self):
```

Comment:
the two tests only vary in the source of keys. Maybe the common code could be put in a single function

Reply:
https://github.com/ARMmbed/mbed-build/pull/37/commits/3efac4a8864309c5733c28b78c5c9ff9f207fef8